frame/nfts/src/functions.rs

```diff
@@ -169,7 +179,12 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		collection: T::CollectionId,
 		item: T::ItemId,
 		owner: T::AccountId,
-		config: ItemConfig,
+		item_config: ItemConfig,
+		deposit_collection_owner: bool,
+		with_details_and_config: impl FnOnce(
```

Comment:
```suggestion
		config_callback: impl FnOnce(
```
Calling a callback from a `try_mutate` is a bit dangerous, since the `try_mutate` can possibly overwrite the changes that the callback made to storage.

You can use normal `get/set` instead of `try_mutate`.

Reply:
That callback was here from the beginning. I've removed it in a prev PR as it became redundant, but now returned it back and found a use-case for it.